### PR TITLE
Read files in alphabetical order when file globbing

### LINKED ISSUE #41934: 02267_file_globs_schema_inference is broken on AARCH64
Run: https://s3.amazonaws.com/clickhouse-test-reports/0/91b70b412043bdda6d7f322ecc0ae94156faa0f3/stateless_tests__aarch64_.html

Error is only found in the [runlog](https://s3.amazonaws.com/clickhouse-test-reports/0/91b70b412043bdda6d7f322ecc0ae94156faa0f3/stateless_tests__aarch64_/runlog.log):
 
```
2022-09-29 11:29:45 02267_file_globs_schema_inference:                                      [ FAIL ] 0.13 sec. - return code: 27
2022-09-29 11:29:45 Expected server error code: 117 but got: 27 (query: select * from file('02267_data*.jsonl') settings schema_inference_use_cache_for_file=0; --{serverError INCORRECT_DATA}).
2022-09-29 11:29:45 Received exception from server (version 22.10.1):
2022-09-29 11:29:45 Code: 27. DB::Exception: Received from localhost:9000. DB::ParsingException. DB::ParsingException: Cannot parse input: expected '{' before: '1\n[1,2,3]\n': While executing JSONEachRowRowInputFormat: While executing File: (in file/uri /var/lib/clickhouse/user_files/02267_data1.jsonl): (at row 1)
2022-09-29 11:29:45 . (CANNOT_PARSE_INPUT_ASSERTION_FAILED)
2022-09-29 11:29:45 (query: select * from file('02267_data*.jsonl') settings schema_inference_use_cache_for_file=0; --{serverError INCORRECT_DATA})
2022-09-29 11:29:45 , result:
2022-09-29 11:29:45 
2022-09-29 11:29:45 1
2022-09-29 11:29:45 \N
2022-09-29 11:29:45 1
2022-09-29 11:29:45 \N
2022-09-29 11:29:45 
2022-09-29 11:29:45 stdout:
2022-09-29 11:29:45 1
2022-09-29 11:29:45 \N
2022-09-29 11:29:45 1
2022-09-29 11:29:45 \N
2022-09-29 11:29:45 
2022-09-29 11:29:45 
2022-09-29 11:29:45 Settings used in the test: --max_insert_threads=0 --group_by_two_level_threshold=1152921504606846976 --group_by_two_level_threshold_bytes=1152921504606846976 --distributed_aggregation_memory_efficient=1 --fsync_metadata=1 --output_format_parallel_formatting=0 --input_format_parallel_parsing=0 --min_chunk_bytes_for_parallel_parsing=11984955 --max_read_buffer_size=907339 --prefer_localhost_replica=0 --max_block_size=14478 --max_threads=38 --optimize_or_like_chain=0 --optimize_read_in_order=1 --read_in_order_two_level_merge_threshold=10 --optimize_aggregation_in_order=1 --aggregation_in_order_max_block_bytes=37993372 --use_uncompressed_cache=1 --min_bytes_to_use_direct_io=169177437 --min_bytes_to_use_mmap_io=0 --local_filesystem_read_method=pread_threadpool --remote_filesystem_read_method=threadpool --local_filesystem_read_prefetch=1 --remote_filesystem_read_prefetch=0 --compile_expressions=1 --compile_aggregate_expressions=0 --compile_sort_description=1 --merge_tree_coarse_index_granularity=18 --optimize_distinct_in_order=0 --optimize_sorting_by_input_stream_properties=1
2022-09-29 11:29:45 
```

There isn't [a single good run in master in the last 30 days](https://play.clickhouse.com/play?user=play#d2l0aCA3MCBhcyB0ZXN0X25hbWVfbGltaXQKU0VMRUNUCiAgaWRlbnRpdHkoY291bnQoKikgYXMgZmFpbHMpIGFzICJGYWlscyIsCiAgaWRlbnRpdHkoY291bnRJZihjaGVja19zdGFydF90aW1lID4gbm93KCkgLSBpbnRlcnZhbCA3IGRheSkgYXMgcmVjZW50KSBhcyAiUmVjZW50IiwKICBtYXgoY2hlY2tfc3RhcnRfdGltZSkgYXMgIkxhc3QiLAogIGFyZ01heChjaGVja19uYW1lLCBjaGVja19zdGFydF90aW1lKSAiQ2hlY2sgTmFtZSIsCiAgaWYobGVuZ3RoKHRlc3RfbmFtZSkgPCB0ZXN0X25hbWVfbGltaXQsIHRlc3RfbmFtZSwgc3Vic3RyKHRlc3RfbmFtZSwgMSwgdGVzdF9uYW1lX2xpbWl0KSB8fCAnLi4uJykgIlRlc3QgTmFtZSIsCiAgJzxhIGhyZWY9IicgfHwgYXJnTWF4KHJlcG9ydF91cmwsIGNoZWNrX3N0YXJ0X3RpbWUpIHx8ICciPicgfHwgYXJnTWF4KHRlc3Rfc3RhdHVzLCBjaGVja19zdGFydF90aW1lKXx8ICc8L2E+JyAiTGluayB0byBSZXBvcnQiCkZST00gY2hlY2tzCldIRVJFCiAgICBjaGVja19zdGFydF90aW1lIGJldHdlZW4gbm93KCkgLSBpbnRlcnZhbCAzMCBkYXkgYW5kIG5vdygpCiAgICBBTkQgY2hlY2tfc3RhdHVzICE9ICdzdWNjZXNzJwogICAgQU5EIHRlc3Rfc3RhdHVzIG5vdCBpbiAoJ09LJywgJ1NLSVBQRUQnLCAnJykKICAgIEFORCBwdWxsX3JlcXVlc3RfbnVtYmVyID0gMAogICAgQU5EIGNoZWNrX25hbWUgTk9UIExJS0UgJyVTUUxhbmNlciUnCiAgICBhbmQgY2hlY2tfbmFtZSBub3QgbGlrZSAnWWFuZGV4JScKICAgIGFuZCBjaGVja19uYW1lIG5vdCBsaWtlICdUZXN0ZmxvdyUnCiAgICBhbmQgdGVzdF9uYW1lID0gJzAyMjY3X2ZpbGVfZ2xvYnNfc2NoZW1hX2luZmVyZW5jZScKLS0gSXQgaXMgaW1wb3J0YW50IG5vdCB0byBncm91cCBhbGwgdGVzdHMgd2l0aCBlbXB0eSB0ZXN0X25hbWUgdG9nZXRoZXIsCi0tIHN1Y2ggYXMgQVNUIEZ1enplciBhbmQgUGVyZm9ybWFuY2UuCmdyb3VwIGJ5IGlmKHRlc3RfbmFtZSAhPSAnJywgdGVzdF9uYW1lLCBjaGVja19uYW1lKSBhcyB0ZXN0X25hbWUKaGF2aW5nIHJlY2VudCA+IDAKb3JkZXIgYnkgcmVjZW50IGRlc2MsIGZhaWxzIGRlc2MK).

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Read files in alphabetical order when file globbing

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

Closes https://github.com/ClickHouse/ClickHouse/issues/41934
Closes https://github.com/ClickHouse/ClickHouse/pull/41877

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
We don't have to read files in alphabetical order.

File: src/Storages/StorageFile.cpp [hunk 1]
```diff
@@ -131,6 +131,7 @@ std::vector<std::string> listFilesWithRegexpMatching(
 {
     std::vector<std::string> result;
     listFilesWithRegexpMatchingImpl(path_for_ls, for_match, total_bytes_to_read, result);
+    std::sort(result.begin(), result.end()); /// Enforce the same order, independently of the order in the filesystem
```
Comment (alexey-milovidov) [inline src/Storages/StorageFile.cpp:134]: We shouldn't pay for what we don't need.

File: tests/queries/0_stateless/02267_file_globs_schema_inference.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
+# Fast test: Requires JSON
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/02267_file_globs_schema_inference.sh:5]: If you're concerned about this test, @Avogar will fix it.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: src/Storages/StorageFile.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Storages/StorageFile.cpp:134]: The comment does not explain "why",

It reads like this:
```
/// Enforce the same order, independently of the order in the filesystem
```
so what?

**Comment by Algunenano** [comment]
Well, the tests do seem to think that files are read in alphabetical order as it happens in bash or posix' glob; but if you think this isn't necessary we can keep it undefined.

**Pull request closed without merging**